xrRender: staged geometry loading (#489)

vTurbine opened the pull request:
These are final changes to switch all static geometry buffers to staging loading. I'd like to highlight one particular part in `FSkinned.cpp` which restores HW skinning for DX1x and GL renderers.

vTurbine committed:
xrRender: Staged geometry loading

- Using HW skinning for GL/DX1x
- Fixed vertices lock size for readback in DX9

vTurbine committed:
xrRenderGL: unused buffer pool removed

vTurbine committed:
xrRender: VertexElement replacement

vTurbine committed:
xrRenderR3: render targets switched to staging geometry buffers

vTurbine committed:
xrRenderR4: render targets switched to staging geometry buffers

vTurbine committed:
xrRender: staging geometry buffers in volumetrics

vTurbine committed:
xrRender: Utils: unused declarations removed

Xottab-DUTY commented on src/Layers/xrRender/FSkinned.cpp:
```diff
@@ -765,27 +578,28 @@ void CSkeletonX_ext::_CollectBoneFaces(Fvisual* V, u32 iBase, u32 iCount)
             BD.AppendFace(ChildIDX, (u16)(idx / 3));
         }
-        V->p_rm_Vertices->Unlock();
+        V->p_rm_Vertices->Unmap(); // ? keep ?
```
To keep or not to keep?

Let me guess: the code is working good, but this is a potential place for optimization?

vTurbine replied:
Yes, you're right. :)
The idea behind this is simple. While loading animated meshes some kinematics setups access their geometry data. It's reasonable to free host heap buffers right after the meshes get processed but I not sure yet where is the best place for it.

Xottab-DUTY replied:
Aha... Ok then. We should note that we can improve this, but I believe, it's better to do this optimization in another PR)

Maybe, @SkyLoaderr can give us a piece of advice on this?

Xottab-DUTY reviewed (approved):
> I'd like to highlight one particular part in FSkinned.cpp which restores HW skinning for DX1x and GL renderers.

And this is awesome! I didn't even know about it)

vTurbine committed:
xrRender: Visuals: better VB pointers managing

No need to have smartptr since the memory will be freed by `_RELEASE` in
base `IRender_Mesh` destructor
